rules/Php55/RegexMatcher.php

```diff
@@ -46,6 +46,10 @@ public function resolvePatternExpressionWithoutEIfFound(Expr $expr): Concat|Stri
                 return null;
             }
 
+            if ($delimiter === '(' && $pattern[strlen($pattern) - 1] === ')') {
```

Comment:
Also, other brackets can be used as delimiters, to avoid having to escape the characters in the expression: https://www.php.net/manual/en/regexp.reference.delimiters.php

Reply:
I add more brackets to handle that https://github.com/rectorphp/rector-src/pull/3037/commits/78553617ca114cc3bf311e8e7cd9f92c28d4e033